### ISSUE TITLE
[[ a != b || a != c ]] is not tautologically true when b and c use pattern matching

### ISSUE THREAD

**Issue by eatnumber1** [koalaman/shellcheck#586]
I've got the following code:
```bash
[[ "$foo" != */foo || "$foo" != /foo/bar/baz/* ]]
```
This triggers [SC2055] which protects against tautological conditionals which are always true. In this case however, bash performs [pattern matching][patmat] on the right hand side of `==` and `!=` operators which means that this can be false. See the section 3.2.4.2 of the Bash [reference manual][bashref-cond] for a primary reference.

[SC2055]: https://github.com/koalaman/shellcheck/wiki/SC2055
[patmat]: https://www.gnu.org/software/bash/manual/bashref.html#Pattern-Matching
[bashref-cond]: https://www.gnu.org/software/bash/manual/bashref.html#Conditional-Constructs

**Comment by koalaman** [comment]
Well caught! Fixed in 7d278c3
